features/FEATURE_BLE/ble/SecurityManager.h

```diff
@@ -14,16 +14,150 @@
+ * allows you to set whether or not the communication channel you are using to transmit the OOB data is
+ * itself secure against MITM protection - this will set the level of the link security achieved using pairing
+ * that uses this data.
+ *
+ * The most secure pairing is provided by Secure Connections which relies on Elliptical Curve Cryptography.
+ * Support for Secure Connections is dependent on both the stack and controller on both sides supporting
+ * it. If either side doesn't support it Legacy Pairing will be used. This is an older standard of pairing.
+ * If higher security is required legacy pairing can be disabled by calling allowLegacyPairing(false);
+ *
+ * How to use
+ *
+ * First thing you need to do is to initialise the manager by calling init() with your chosen settings.
+ *
+ * The SecurityManager communicates with your application through events. These will trigger calls in
+ * the EventHandler which you must provide by calling the setSecurityManagerEventHandler() function.
+ *
+ * The most important process is pairing. This may be triggered manually by calling requestPairing() or
+ * may be called as a result of the application requiring encryption by calling setLinkEncryption() or
+ * as a result of the application requiring MITM protection through requestAuthentication().
+ *
+ * All these can be implicitly called by using setLinkSecurity() to conveniently set the required
+ * security for the link. The SecurityManager will trigger all the process required to achieve the set
+ * security level. Security level can only be escalated. Asking the Security Manager for a lower
+ * security level than the existing one will not fail but will result in a event informing the
+ * application through linkEncryptionResult() of the current level (which remains unchanged).
+ *
+ * Depending on the IO capabilities and OOB usage settings different pairing algorithms will be chosen.
+ * They will produce appropriate events which must be handled by your EventHandler. If your event handler
+ * doesn't support all the calls you must not set IO capabilities or set OOB usage in such a way that would
+ * trigger them or else the pairing will fail (usually by timing out).
+ *
+ * The simplest example is a pairing of a device with no IO capabilities and no OOB data available.
+ * With such limited pairing capabilities the "just works" method will be employed. This does not provide
+ * any MITM protection. The pairing (triggered implicitly or called explicitly) will result in an event
+ * being generated on the peer calling pairingRequest(). The event handler must make a decision (either in
+ * the application itself or based on user interaction) whether to accept the pairing and call
+ * accetPairing() or cancelPairing(). The result will be communicated on both peers through an event calling
+ * pairingResult() in the EventHandler.
+ *
+ *
+ * Sequence diagram "Just Works" pairing
+ *
```

Comment:
It might be interesting to surround diagrams with `verbatim` and `endverbatim` tags: https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdverbatim

Reply:
surrounded